corehq/apps/users/models.py

```diff
@@ -724,16 +724,29 @@ def migrate_eula(cls, data):
     def is_eula_signed(self, version=CURRENT_VERSION):
         if self.is_superuser:
             return True
-        for eula in self.eulas:
-            if eula.version == version:
-                return eula.signed
+        current_domain = getattr(self, 'current_domain', None)
+        current_eula = self.eula
+        if current_eula.version == version:
+            if toggles.FORCE_ANNUAL_TOS.enabled(current_domain):
+                elapsed = datetime.now() - current_eula.date if current_eula.date else 365
+                return current_eula.signed and elapsed.days < 365
+            return current_eula.signed
         return False
 
     def get_eula(self, version):
+        current_eula = None
         for eula in self.eulas:
             if eula.version == version:
-                return eula
-        return None
+                if not current_eula or eula.date > current_eula.date:
```

Comment:
Won't this always evaluate as true since current_eula is set to None?

Reply:
at first it's set to None. There is a scenario where current_eula is set to eula because the version matches, but later in the loop current_eula has an older date and should be reassigned to the eula that matches the version and has a newer date.

Reply:
oh duh, I missed the reassignment line